Shared/Characters/GhostBase.gd

```diff
@@ -43,27 +57,56 @@ func _physics_process(_delta):
 
 # Stops the ghost and triggers the base classes hit function
 # OVERRIDE #
-func hit() -> void:
+func hit(perpetrator) -> void:
 	_is_playing = false
-	.hit()
-
+	.hit(perpetrator)
 
+# OVERRIDE #
+func quiet_hit(perpetrator) -> void:
+	_is_playing = false
+	.quiet_hit(perpetrator)
+	
 # Starts moving the ghost and enables collision
 func start_playing(start_time: int) -> void:
 	_is_playing = true
-	_collision_shape.disabled = false
 	_current_frame_index = 0
 	_start_time = start_time
 
 
 # Stops moving the ghost and disables collision
 func stop_playing() -> void:
 	_is_playing = false
-	_collision_shape.disabled = true
 
 
 # Applies the given frame
 func _apply_record_frame(record_frame: RecordFrame):
 	.set_position(record_frame.position)
 	.set_rotation_y(record_frame.rotation_y)
 	.trigger_actions(record_frame.buttons)
+
+
+func enable_body():
+	if not is_record_data_set():
+		return
```

Comment:
Right, this does happen because `GhostManager` does not check for set records when enabling all ghosts while calling `refresh_active_ghosts` and this happens as soon as the first prep phase starts, but I agree actually checking in GhostManager is cleaner. Will replace it with an assert.